### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by mforets**
Minor docs tweaks for `@system` and some other related functions in `macros.jl`.

### CODE REVIEW

File: src/macros.jl [hunk 1]
```diff
@@ -595,69 +602,94 @@ end
...
+- set constraints, e.g. `x ∈ X`
+- input constraints, e.g. `u ∈ U`
+- dimensionality, e.g. `dim: (2,1)` or `dim = 1`
+- specification of the input variable, e.g. `input: u` or `input = u`
+- specification of the noise variable, e,g, `noise: w` or `noise = w`
 
-If we want to change the default name of the input or noise variable, this can be
-done by adding the term `input: var` where `var` corresponds to the new name of
-the input variable.
+The macro call should is then formed by separating the previous sub-expressions
```
Comment (schillic) [inline src/macros.jl]:
```suggestion
The macro call is then formed by separating the previous sub-expressions
```

File: src/macros.jl [hunk 2]
```diff
@@ -595,69 +602,94 @@ end
...
+any particular order. Moreover, the only mandatory term is the dynamic equation;
+the other terms are optional and default values may apply depending on the system
+type; this is explained next.
+
+**Dynamic equation.** The time derivative in a continuous system is specified
+by using `'`, as in `x' = A*x`. A discrete system is specified using `⁺` (which
+can be written with the combination of keys `\\^+[TAB]`), as in `x⁺ = A*x`.
+Moreover, the asterisk denoting matrix-vector products is optional. For instance,
+both `x' = Ax` and `x' = A*x` are parsed as the linear continuous system whose
+state matrix is `A` (matrix that is supposed to be defined at the call site).
```
Comment (schillic) [inline src/macros.jl]:
```suggestion
state matrix is `A`. The matrix is supposed to be defined at the call site.
```

File: src/macros.jl [hunk 3]
```diff
@@ -595,69 +602,94 @@ end
...
+  equation is parsed as a controlled linear (affine) system with input `foo`.
+  Note that in this case, `input` variable does not correspond to the default
+  value of `u`, but `foo` is parsed as being the input.
+
+- If the left-hand side contains a multiplicative term in the form `E*x⁺` or `E*x'`,
+  the equation is parsed as an algebraic system. In this case, the asterisk
+  `*` operator is mandatory.
+
+- Systems of the form `x' = α*x` where `α` is a scalar are parsed as linear
+  systems. The default dimension is `1`; if the system is higher-dimensional use
```
Comment (schillic) [inline src/macros.jl]:
```suggestion
  systems. The default dimension is `1`; if the system is higher-dimensional, use
```

File: src/macros.jl [hunk 4]
```diff
@@ -595,69 +602,94 @@ end
...
 julia> A = [1. 0; 0 1.];
 
 julia> @system(x' = A*x)
 LinearContinuousSystem{Float64,Array{Float64,2}}([1.0 0.0; 0.0 1.0])
 ```
-A discrete system can be defined by using `⁺`:
 
-```jldoctest
-julia> A = [1. 0; 0 1.];
+A discrete system can be defined by using `⁺`:
```
Comment (schillic) [inline src/macros.jl]:
```suggestion
A discrete system is defined by using `⁺`:
```
"can be" suggests that there are alternatives.

**Comment by mforets** [comment]
Thanks for the reviews!

**Pull request merged**